File: app/src/main/java/com/stathis/elmepaunivapp/ui/syllabus/model/Semester.java
package com.stathis.elmepaunivapp.ui.syllabus.model;

import java.util.List;

public class Semester extends Object {

    String semester;
    String lessonInfo;
    List<Lesson> lessons;

    public Semester(String semester, String lessonInfo, List<Lesson> lessons) {
        this.semester = semester;
        this.lessonInfo = lessonInfo;
        this.lessons = lessons;
    }

    public String getSemester() {
        return semester;
    }

    public void setSemester(String semester) {
        this.semester = semester;
    }

    public String getLessonInfo() {
        return lessonInfo;
    }

    public void setLessonInfo(String lessonInfo) {
        this.lessonInfo = lessonInfo;
    }

    public List<Lesson> getLessons() {
        return lessons;
    }

    public void setLessons(List<Lesson> lessons) {
        this.lessons = lessons;
    }
}

File: app/src/main/java/com/stathis/elmepaunivapp/ui/syllabus/model/Semester.kt
package com.stathis.elmepaunivapp.ui.syllabus.model

import android.os.Parcelable
import com.stathis.elmepaunivapp.abstraction.LocalModel
import com.stathis.elmepaunivapp.ui.syllabus_lessons.model.Lesson
import kotlinx.android.parcel.Parcelize
import java.io.Serializable

@Parcelize
data class Semester(

    val semester: String,
    val lessonInfo: String,
    val lessons: List<Lesson>

) : LocalModel, Parcelable {
    override fun equalsContent(obj: LocalModel): Boolean = when(obj){
        is Semester -> semester == obj.semester && lessonInfo == obj.lessonInfo && lessons == obj.lessons
        else -> false
    }
}